### DISCUSSION TITLE
daly bms via uart to deye inverter with can

### DISCUSSION THREAD

**Discussion by k4ru44** [ai-republic/bms-to-inverter#8] [Q&A] [answered]
hello, i have found your project and im curious if it is possible to connect together my daly bms (which has only uart communication port for pc) with my new daly inverter that used can or rs485 connection?

**Comment by ai-republic** [comment]
Hi,
yes I'm sure you can. 
Could you tell me more about your environment? Will you be using a Raspberry PI/Linux or Windows?
Please note that at the moment CAN protocol is only supported on Linux. But RS485 is supported for Windows.
Please also let me know which Daly inverter you have.

**Reply by k4ru44** [reply]
actually im starting with new hardware and it will be deye sun-8k-sg04lp3-eu.

i have no idea which envoirnment i should use for that - probably raspberry is the best choice and it could be used and work 24/7 against other options. but i have never done anything like that before - if it will work then i can try to figure it out, especially cause i have perfectly working daly bms and brand new active balancer for it..

**Reply by kommando828** [reply]
A pi is a good platform but you will have an initial steep learning curve if you are used to Windows and not Linux. That's where I was 12 months ago, but if you work at it then you pick it up quickly. A Pi4 will give you the best bang for buck currently, the Pi5 is a bit new and there are limited Operating images for it. I use Buster Raspbian image as the initial sign on is easy, later ones are encrypted and for a headless install caused me issues. Have a screen, keyboard and mouse for working on it at the start but when its working as you want you can go headless ie not screen etc and SSH in from a PC to make any changes. That will also reduce the CPU load. Once you have a stable setup back up the SD card and keep the copy SD in a safe place. With a copied SD card you can be up and running in 5 mins if the first SD is eaten by the Pi (a known issue with Pi's)  I have had a Pi that ate SD's every few days, all my recent ones have caused no issues at all.

**Reply by ai-republic** [reply, marked as answer]
Hi,
yes, I would suggest a Raspberry PI 4B with 4 or better 8GB.
As @kommando828 said it's the best value for money and will serve adequately.

But @kommando828 also makes it sound very complicated to a newbie. It is actually very simple and I can guide you setting it up in a few minutes.

So let me understand your system setup a litte better, then I can make recommendations on which hardware works well.
Your BMS is a Daly with only UART enabled with Daly 1A active balancer. Your inverter is a Deye Sun-8k-SG04LP3-EU supporting both RS4885 and CAN. Did I understand that correct?

**Reply by k4ru44** [reply]
i can try do it on pi 4 if its easier and have best money/value ratio. 

@ai-republic yes. i have basic daly bms 150A with only UART protocol - now i have both BT and WIFI modules for it. inverter is as you said.

**Reply by ai-republic** [reply]
Then I would recommend a Raspberry PI 4B 8Gb and a [Waveshare RS485/CAN hat](https://www.waveshare.com/rs485-can-hat.htm).
They are easy to set up and reliable.

**Comment by k4ru44** [comment]
hello again, im about to buy all necessary stuff to start with my project, but im thinking about possibility to use Pi for home assistant as well - is it possible? by docker or something like that to run both images in pi?

**Reply by ai-republic** [reply]
Yes, you can install the home assistant Raspberry image and then install the Terminal and SSH to connect to the Home Assistant OS and then perform the installation as decribed in the Wiki. 
Important is, that you install Java first by calling:
```
sudo apt install default-jdk
```
